src/main/java/com/microsoft/sqlserver/jdbc/Util.java

```diff
@@ -478,27 +465,27 @@ else if (ch == ':')
         // Exit
         switch (state) {
             case inServerName:
-                result = result.trim();
-                if (result.length() > 0) {
+                String property = result.toString().trim();
+                if (property.length() > 0) {
                     if (logger.isLoggable(Level.FINE)) {
-                        logger.fine("Property:serverName " + "Value:" + result);
+                        logger.fine("Property:serverName " + "Value:" + property);
                     }
-                    p.put(SQLServerDriverStringProperty.SERVER_NAME.toString(), result);
+                    p.put(SQLServerDriverStringProperty.SERVER_NAME.toString(), property);
                 }
                 break;
             case inPort:
-                result = result.trim();
+                property = result.toString().trim();
```

Comment:
Lines 477 and 484 will probably error out, because `property` variable is actually not defined here. You should declare property as a String every time, or move up the declaration of line 468 to somewhere above.

Reply:
odd, I'm surprised it's not complaining about that

Reply:
It doesn't error out, it's declared in the first switch statement.

Reply:
But there is a `break;` after the inServerName case. I think the tests are not failing because every test case had `inServerName` get called before other cases. Can you test this method by providing it a connection string that has maybe the port number come before the server name?